### PR TITLE
Added Plus Button to Select Asset Views

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Added Plus Button to Select Asset Views

1) Adds a `+` icon button to the top right corner of the `SelectAsset` views in the `Buy/Send/Swap` widget and `Panel`
2) `EditVisibleAssets` modal is now it's own route `brave://wallet/crypto/portfolio/add-asset`

Resolves <https://github.com/brave/brave-browser/issues/19139>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/145652322-d3af36ec-7290-4233-8102-95b8488edaba.mov

### CODE REVIEW

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts [hunk 1]
```diff
@@ -387,7 +387,7 @@ handler.on(PanelActions.signMessageHardware.getType(), async (store, messageData
   }
   const payload: SignMessageHardwareProcessedPayload =
     signed.success ? { success: signed.success, id: messageData.id, signature: signed.payload }
-                   : { success: signed.success, id: messageData.id, error: signed.error }
+      : { success: signed.success, id: messageData.id, error: signed.error }
```
Comment (onyb) [inline components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts:390]: Let's follow this style.

```
condition
  ? then
  : else
```

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts [hunk 2]
```diff
@@ -467,6 +467,14 @@ handler.on(PanelActions.expandWalletAccounts.getType(), async (store) => {
   })
 })
 
+handler.on(PanelActions.expandWalletAddAsset.getType(), async (store) => {
+  chrome.tabs.create({ url: 'chrome://wallet/crypto/portfolio/add-asset' }, () => {
```
Comment (onyb) [inline components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts:471]: Is there a constant for this route?

File: components/brave_wallet_ui/page/container.tsx [hunk 3]
```diff
@@ -463,6 +464,18 @@ function Container (props: Props) {
     props.walletActions.expandWalletNetworks()
   }
 
+  const onShowVisibleAssetsModal = (value: boolean) => {
```
Comment (onyb) [inline components/brave_wallet_ui/page/container.tsx:467]: Let's add a comment explaining the purpose of `value`.

**Review by onyb — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts]: Oh weird! My code formatter must have picked up on this when I saved my changes in this file.
No related to this PR, but fixed the code styling any way. 😎

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts]: There is a partial constant for these routes that doesn't include `chrome://wallet` 
Updated it to use the second half, as well as others.

File: components/brave_wallet_ui/page/container.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/container.tsx]: Changed `value` too `showModal` to give more clarification!

**Review by onyb — APPROVED** [review]

**Pull request merged**